specifications/api/swagger.yaml

```diff
@@ -1851,6 +1959,85 @@ components:
         state: *walletState
         tip: *blockReference
 
+    ApiActiveSharedWallet: &ApiActiveSharedWallet
+      type: object
+      required:
+        - id
+        - name
+        - account_public_key
+        - account_ix
+        - address_pool_gap
+        - payment_script_template
+        - balance
+        - delegation
+        - state
+        - tip
+      properties:
+        id: *walletId
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_ix: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+        balance: *walletBalance
+        delegation: *ApiWalletDelegation
+        state: *walletState
+        tip: *blockReference
+
+    ApiPendingSharedWallet: &ApiPendingSharedWallet
+      type: object
+      required:
+        - id
+        - name
+        - account_public_key
+        - account_ix
+        - address_pool_gap
+        - payment_script_template
+      properties:
+        id: *walletId
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_ix: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
+        payment_script_template: *scriptTemplate
+        delegation_script_template: *scriptTemplate
+
+    ApiSharedWallet: &ApiSharedWallet
+      type: object
+      oneOf:
+        - title: Pending shared wallet (collecting account public keys for each co-signer)
+          <<: *ApiPendingSharedWallet
+        - title: Active shared wallet (account public keys for each co-signers is collected)
+          <<: *ApiActiveSharedWallet
+
+    ApiSharedWalletPostData: &ApiSharedWalletPostData
+      type: object
+      required:
+        - name
+        - account_public_key
+        - account_ix
+        - address_pool_gap
+        - payment_script_template
+      properties:
+        name: *walletName
+        account_public_key: *walletAccountPubkey
+        account_ix: *derivationSegment
+        address_pool_gap: *walletAddressPoolGap
```

Comment:
I am not sure whether we want to allow setting this one through the API :no_mouth: ... because if co-signers messes up with that and define different gaps, it may corrupt the wallet for others. Unless strictly required by product, I'd leave that one out of the API for now and use a sensible default internally.

Reply:
in order to handle both ways of retrieving (from mnemonic and account public key) I used `ApiWalletOrAccountPostData` and there is pool gap  to be specified. The advantage is that we can reuse existent battle-field code